src/metadata.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use grpc_sys::{self, GrpcMetadataArray};
+use std::{ptr, slice, str};
+use std::borrow::Cow;
+
+use libc;
+
+use error::{Error, Result};
+
+fn normalize_key(key: &str, binary: bool) -> Result<Cow<str>> {
```

Comment:
Is this required by gRPC? Any reference?

Reply:
See docs at https://github.com/pingcap/grpc-rs/pull/161/files#diff-26aa3c8895057aa93d60e92c0dee6ddcR145